superset-frontend/src/views/components/MenuRight.tsx

```diff
@@ -144,9 +158,30 @@ const RightMenu = ({
       {menu.label}
     </>
   );
+
+  const handleMenuSelection = (itemChose: React.Key) => {
+    if (itemChose === GlobalMenuDataOptions.DB_CONNECTION) {
+      setShowModal(true);
+      setEngine('');
```

Comment:
I tried moving this logic to the into component, but since we are managing the state in the upper component (MenuRight) it doesn't reset. 

Open to other strategies here, but was having a hard time getting it to work without reseting from the upper component

Reply:
I think it's ok to leave the logic here for which engine was chosen, but maybe instead of clearing it out on menu selection, we can clear out the engine value on the dbmodal onHide event?